Src/Test.UnitTests.SarifPatternMatcher/SearchSkimmerTests.cs

```diff
@@ -15,19 +15,35 @@ namespace Microsoft.CodeAnalysis.SarifPatternMatcher.SarifPatternMatcher
 {
     public class SearchSkimmerTests
     {
-        private static MatchExpression CreateGuidDetectingMatchExpression()
+        private static MatchExpression CreateGuidDetectingMatchExpression(bool denyRegexEnabled = false)
         {
             const string guidRegexText = "(?i)[0-9a-f]{8}[-]?([0-9a-f]{4}[-]?){3}[0-9a-f]{12}";
 
             return new MatchExpression
             {
                 MatchLengthToDecode = Guid.NewGuid().ToString().Length,
                 ContentsRegex = guidRegexText,
+                FileNameDenyRegex = denyRegexEnabled ? "(?i)\\.asc$" : ""
             };
         }
 
+        private static MatchExpression CreateFileDetectingMatchExpression()
+        {
+            const string fileNameRegexText = "(?i)\\.asc$";
+
+            return new MatchExpression
+            {
+                FileNameAllowRegex = fileNameRegexText
+            };
+        }
+
+        private static MatchExpression CreateEmptyMatchExpression()
+        {
+            return new MatchExpression();
+        }
+
         [Fact]
-        public void SearchSkimmer_DetectsBase64EncodedPattern()
```

Comment:
>SearchSkimmer_DetectsBase64EncodedPattern [](start = 20, length = 41)

This is our test naming convention to put the name of the type under test as a prefix. it helps with search in the test explorer. You probably don't like it because of style rules that flag the underscore?? :) we should have a stylecop ruleset specifically for test code that turns this one off.

Reply:
I just think that that is strange. Today, it's organized like this:
1. project name
  - namespace
     - class name
         - test name

So, in theory, its a duplicated work, since test name = classname_testname. Either way, both works :)

---
In reply to: [546093717](https://github.com/microsoft/sarif-pattern-matcher/pull/44#discussion_r546093717) [](ancestors = 546093717)